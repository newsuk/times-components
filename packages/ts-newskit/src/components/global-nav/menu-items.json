{
    "mainMenuItems": [
      {
        "title": "Home",
        "url": "/home",
        "slug": "home"
      },
      {
        "title": "News",
        "url": "/news",
        "slug": "news",
        "items": [
          {
            "title": "Top stories",
            "url": "/news",
            "slug": "top-stories-news"
          },
          {
            "title": "UK",
            "url": "/uk",
            "slug": "uk"
          },
          {
            "title": "World",
            "url": "/world",
            "slug": "world"
          },
          {
            "title": "Health & Science",
            "url": "/health-and-science",
            "slug": "health-and-science"
          },
          {
            "title": "Environment",
            "url": "/environment",
            "slug": "environment"
          },
          {
            "title": "Royal family",
            "url": "/royal-family",
            "slug": "royal-family"
          },
          {
            "title": "Investigations",
            "url": "/investigations",
            "slug": "investigations"
          },
          {
            "title": "Obituaries",
            "url": "/obituaries",
            "slug": "obituaries"
          },
          {
            "title": "Education",
            "url": "/education",
            "slug": "education"
          },
          {
            "title": "Scotland",
            "url": "/scotland",
            "slug": "scotland"
          },
          {
            "title": "Law",
            "url": "/law",
            "slug": "law"
          }
        ]
      },
      {
        "title": "Comment",
        "url": "/comment",
        "slug": "comment",
        "items": [
          {
            "title": "Top stories",
            "url": "/comment",
            "slug": "top-stories-comment"
          },
          {
            "title": "Columnists",
            "url": "/columnists",
            "slug": "columnists"
          },
          {
            "title": "The Times view",
            "url": "/the-times-view",
            "slug": "the-times-view"
          },
          {
            "title": "Letters to the Editor",
            "url": "/letters-to-the-editor",
            "slug": "letters-to-the-editor"
          },
          {
            "title": "Cartoons",
            "url": "/cartoons",
            "slug": "cartoons"
          },
          {
            "title": "Register",
            "url": "/register",
            "slug": "register"
          }
        ]
      },
      {
        "title": "Life & Style",
        "url": "/life-and-style",
        "slug": "life-and-style",
        "items": [
          {
            "title": "Top stories",
            "url": "/life-and-style",
            "slug": "top-stories-life-and-style"
          },
          {
            "title": "Property",
            "url": "/property",
            "slug": "property"
          },
          {
            "title": "Sex & Relationships",
            "url": "/sex-and-relationships",
            "slug": "sex-and-relationships"
          },
          {
            "title": "Food & Drink",
            "url": "/food-and-drink",
            "slug": "food-and-drink"
          },
          {
            "title": "Travel",
            "url": "/travel",
            "slug": "travel"
          },
          {
            "title": "Fashion",
            "url": "/fashion",
            "slug": "fashion"
          },
          {
            "title": "Beauty",
            "url": "/beauty",
            "slug": "beauty"
          },
          {
            "title": "Fitness & Wellbeing",
            "url": "/fitness-and-wellbeing",
            "slug": "fitness-and-wellbeing"
          },
          {
            "title": "Luxury",
            "url": "/luxury",
            "slug": "luxury"
          },
          {
            "title": "Driving",
            "url": "/driving",
            "slug": "driving"
          },
          {
            "title": "Parenting",
            "url": "/parenting",
            "slug": "parenting"
          }
        ]
      },
      {
        "title": "Arts & Books",
        "url": "/arts-and-books",
        "slug": "arts-and-books",
        "items": [
          {
            "title": "Top stories",
            "url": "/arts-and-books",
            "slug": "top-stories-arts-and-books"
          },
          {
            "title": "TV & Radio",
            "url": "/tv-and-radio",
            "slug": "tv-and-radio"
          },
          {
            "title": "Books",
            "url": "/books",
            "slug": "books"
          },
          {
            "title": "Music",
            "url": "/music",
            "slug": "music"
          },
          {
            "title": "Film",
            "url": "/film",
            "slug": "Theatre"
          },
          {
            "title": "Opera",
            "url": "/opera",
            "slug": "opera"
          },
          {
            "title": "Arts",
            "url": "/arts",
            "slug": "arts"
          }
        ]
      },
      {
        "title": "Sport",
        "url": "/sport",
        "slug": "sport",
        "items": [
          {
            "title": "Top stories",
            "url": "/sport",
            "slug": "top-stories-sport"
          },
          {
            "title": "Football",
            "url": "/football",
            "slug": "football"
          },
          {
            "title": "Rugby Union",
            "url": "/rugby-union",
            "slug": "rugby-union"
          },
          {
            "title": "Tennis",
            "url": "/tennis",
            "slug": "tennis"
          },
          {
            "title": "Cricket",
            "url": "/cricket",
            "slug": "cricket"
          },
          {
            "title": "Athletics",
            "url": "/athletics",
            "slug": "athletics"
          },
          {
            "title": "Cycling",
            "url": "/cycling",
            "slug": "cycling"
          },
          {
            "title": "Formula One",
            "url": "/formula-one",
            "slug": "formula-one"
          },
          {
            "title": "Golf",
            "url": "/golf",
            "slug": "golf"
          },
          {
            "title": "Horse Racing",
            "url": "/horse-racing",
            "slug": "horse-racing"
          },
          {
            "title": "Boxing",
            "url": "/boxing",
            "slug": "boxing"
          }
        ]
      },
      {
        "title": "Business & Money",
        "url": "/business-and-money",
        "slug": "business-and-money",
        "items": [
          {
            "title": "Top stories",
            "url": "/business-and-money",
            "slug": "top-stories-business-and-money"
          },
          {
            "title": "Money",
            "url": "/money",
            "slug": "money"
          },
          {
            "title": "Economics",
            "url": "/economics",
            "slug": "economics"
          },
          {
            "title": "Markets",
            "url": "/markets",
            "slug": "markets"
          },
          {
            "title": "Technology",
            "url": "/technology",
            "slug": "technology"
          },
          {
            "title": "Entrepreneurs",
            "url": "/entrepreneurs",
            "slug": "entrepreneurs"
          },
          {
            "title": "Companies",
            "url": "/companies",
            "slug": "companies"
          }
        ]
      },
      {
        "title": "Puzzles",
        "url": "/puzzles",
        "slug": "puzzles",
        "items": [
          {
            "title": "Crossword Club",
            "url": "/crossword-club",
            "slug": "crossword-club"
          },
          {
            "title": "Crosswords",
            "url": "/crosswords",
            "slug": "crosswords"
          },
          {
            "title": "Mind games",
            "url": "/mind-games",
            "slug": "mind-games"
          },
          {
            "title": "Daily Quiz",
            "url": "/daily-quiz",
            "slug": "daily-quiz"
          },
          {
            "title": "Soduku",
            "url": "/soduku",
            "slug": "soduku"
          }
        ]
      },
      {
        "title": "Magazines",
        "url": "/magazines",
        "slug": "magazines",
        "items": [
          {
            "title": "The Sunday Times magazine",
            "url": "/the-sunday-times-magazine",
            "slug": "the-sunday-times-magazine"
          },
          {
            "title": "Culture magazine",
            "url": "/culture-magazine",
            "slug": "culture-magazine"
          },
          {
            "title": "Weekend magazine",
            "url": "/weekend-magazine",
            "slug": "weekend-magazine"
          },
          {
            "title": "Style magazine",
            "url": "/style-magazine",
            "slug": "style-magazine"
          },
          {
            "title": "The Times magazine",
            "url": "/the-times-magazine",
            "slug": "the-times-magazine"
          }
        ]
      }
    ],
    "moreMenuItems": [
      {
        "title": "E-paper",
        "url": "/e-paper",
        "slug": "e-paper"
      },
      {
        "title": "Newsletters",
        "url": "/newsletters",
        "slug": "newsletters"
      },
      {
        "title": "TV Guide",
        "url": "/tv-guide",
        "slug": "tv-guide"
      },
      {
        "title": "Times +",
        "url": "/times-plus",
        "slug": "times-plus"
      },
      {
        "title": "Times Radio",
        "url": "/times-radio",
        "slug": "times-radio"
      },
      {
        "title": "Times Podcasts",
        "url": "/podcasts",
        "slug": "podcasts"
      },
      {
        "title": "Times Money Mentor",
        "url": "/money-mentor",
        "slug": "money-mentor"
      },
      {
        "title": "Times Travel",
        "url": "/times-travel",
        "slug": "times-travel"
      },
      {
        "title": "Times wine club",
        "url": "/wine-club",
        "slug": "wine-club"
      },
      {
        "title": "Times appointments",
        "url": "/times-appointments",
        "slug": "times-appointments"
      },
      {
        "title": "The Sunday Times Rich List",
        "url": "/rich-list",
        "slug": "rich-list"
      },
      {
        "title": "The Times good uni guide",
        "url": "/uni-guide",
        "slug": "uni-guide"
      },
      {
        "title": "The Times good school guide",
        "url": "/school-guide",
        "slug": "school-guide"
      },
      {
        "title": "The Times best places to live",
        "url": "/best-places-to-live",
        "slug": "best-places-to-live"
      },
      {
        "title": "The Times best places to stay",
        "url": "/best-places-to-stay",
        "slug": "best-places-to-stay"
      },
      {
        "title": "Growth 100",
        "url": "/growth-100",
        "slug": "growth-100"
      }
    ],
    "accountMenuItems": [
      {
        "title": "Manage account",
        "url": "/account",
        "slug": "manage-account"
      },
      {
        "title": "Change password",
        "url": "/change-password",
        "slug": "change-password"
      },
      {
        "title": "My Newsletters",
        "url": "/my-newsletters",
        "slug": "my-newsletters"
      },
      {
        "title": "My articles",
        "url": "/my-articles",
        "slug": "my-articles"
      },
      {
        "title": "Contact us",
        "url": "/contact-us",
        "slug": "contact-us"
      },
      {
        "title": "Log out",
        "url": "/log-out",
        "slug": "log-out"
      }
    ]
  }