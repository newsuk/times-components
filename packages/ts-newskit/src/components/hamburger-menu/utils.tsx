import React from 'react';
import { Story } from '@storybook/addon-docs';
import { ThemeProvider } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

export const StoryWithTheme: React.FC<{ name: string }> = ({
  name,
  children
}) => (
  <ThemeProvider theme={TimesWebLightTheme}>
    <Story name={name}>{children}</Story>
  </ThemeProvider>
);
