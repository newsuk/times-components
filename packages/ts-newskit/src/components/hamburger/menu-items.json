{
  "menuItems": [
    {
      "title": "Home",
      "url": "/home",
      "slug": "home"
    },
    {
      "title": "News",
      "url": "/news",
      "slug": "news",
      "items": [
        {
          "title": "News 1",
          "url": "/news-1",
          "slug": "news-1"
        },
        {
          "title": "News 2",
          "url": "/news-2",
          "slug": "news-2"
        },
        {
          "title": "News 3",
          "url": "/news-3",
          "slug": "news-3"
        }
      ]
    },
    {
      "title": "Comment",
      "url": "/comment",
      "slug": "comment",
      "items": [
        {
          "title": "Comment 1",
          "url": "/comment-1",
          "slug": "comment-1"
        },
        {
          "title": "Comment 2",
          "url": "/comment-2",
          "slug": "comment-2"
        },
        {
          "title": "Comment 3",
          "url": "/comment-3",
          "slug": "comment-3"
        }
      ]
    },
    {
      "title": "Life & Style",
      "url": "/life-and-style",
      "slug": "life-and-style",
      "items": [
        {
          "title": "Fashion",
          "url": "/fashion",
          "slug": "fashion"
        },
        {
          "title": "Food & Drink",
          "url": "/food-and-drink",
          "slug": "food-and-drink"
        }
      ]
    },
    {
      "title": "Sport",
      "url": "/sport",
      "slug": "sport",
      "items": [
        {
          "title": "Football",
          "url": "/football",
          "slug": "football"
        },
        {
          "title": "Rugby Union",
          "url": "/rugby-union",
          "slug": "rugby-union"
        },
        {
          "title": "Tennis",
          "url": "/tennis",
          "slug": "tennis"
        },
        {
          "title": "Cricket",
          "url": "/cricket",
          "slug": "cricket"
        },
        {
          "title": "Athletics",
          "url": "/athletics",
          "slug": "athletics"
        },
        {
          "title": "Cycling",
          "url": "/cycling",
          "slug": "cycling"
        },
        {
          "title": "Formula One",
          "url": "/formula-one",
          "slug": "formula-one"
        },
        {
          "title": "Golf",
          "url": "/golf",
          "slug": "golf"
        },
        {
          "title": "Horse Racing",
          "url": "/horse-racing",
          "slug": "horse-racing"
        },
        {
          "title": "Boxing",
          "url": "/boxing",
          "slug": "boxing"
        }
      ]
    },
    {
      "title": "Business & Money",
      "url": "/business-and-money",
      "slug": "business-and-money",
      "items": [
        {
          "title": "Business 1",
          "url": "/business-1",
          "slug": "business-1"
        },
        {
          "title": "Business 2",
          "url": "/business-2",
          "slug": "business-2"
        }
      ]
    },
    {
      "title": "Puzzles",
      "url": "/puzzles",
      "slug": "puzzles",
      "items": [
        {
          "title": "Puzzle Club",
          "url": "/puzzle-club",
          "slug": "puzzle-club"
        },
        {
          "title": "Crossword",
          "url": "/crossword",
          "slug": "crossword"
        }
      ]
    }
  ],
  "moreMenuItems": [
    {
      "title": "E-paper",
      "url": "/e-paper",
      "slug": "e-paper"
    },
    {
      "title": "Newsletters",
      "url": "/newsletters",
      "slug": "newsletters"
    },
    {
      "title": "Magazine",
      "url": "/magazine",
      "slug": "magazine"
    },
    {
      "title": "TV Guide",
      "url": "/tv-guide",
      "slug": "tv-guide"
    },
    {
      "title": "Times +",
      "url": "/times-plus",
      "slug": "times-plus"
    },
    {
      "title": "Times Radio",
      "url": "/times-radio",
      "slug": "times-radio"
    },
    {
      "title": "Podcasts",
      "url": "/podcasts",
      "slug": "podcasts"
    },
    {
      "title": "Money Mentor",
      "url": "/money-mentor",
      "slug": "money-mentor"
    },
    {
      "title": "Times Travel",
      "url": "/times-travel",
      "slug": "times-travel"
    },
    {
      "title": "Wine Club",
      "url": "/wine-club",
      "slug": "wine-club"
    },
    {
      "title": "Rich List",
      "url": "/rich-list",
      "slug": "rich-list"
    },
    {
      "title": "Uni Guide",
      "url": "/uni-guide",
      "slug": "uni-guide"
    },
    {
      "title": "School Guide",
      "url": "/school-guide",
      "slug": "school-guide"
    },
    {
      "title": "Best places to live",
      "url": "/best-places-to-live",
      "slug": "best-places-to-live"
    },
    {
      "title": "Best places to stay",
      "url": "/best-places-to-stay",
      "slug": "best-places-to-stay"
    },
    {
      "title": "Growth 100",
      "url": "/growth-100",
      "slug": "growth-100"
    },
    {
      "title": "Announcements",
      "url": "/announcements",
      "slug": "announcements"
    }
  ]
}