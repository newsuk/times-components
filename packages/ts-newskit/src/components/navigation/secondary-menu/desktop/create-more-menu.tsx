import React, { Fragment } from 'react';
import { MenuItem } from 'newskit';
import { SecondaryMenuOptions, SecondaryMenuItem } from '../types';
import { MenuDividerDropdown } from '../styles';

export const CreateMoreMenu: React.FC<{
  options: SecondaryMenuOptions;
  data: SecondaryMenuItem[];
  moreMenuItemsLength: number;
  clickHandler: (title: string) => void;
}> = ({ options, data, moreMenuItemsLength, clickHandler }) => {
  const { handleSelect, isSelected, isExpanded, setIsExpanded } = options;

  return (
    <>
      {data.slice(-moreMenuItemsLength).map(({ title, url }) => (
        <Fragment key={title}>
          <MenuItem
            href={url}
            overrides={{
              stylePreset: 'menuItemMore',
              typographyPreset: 'newPreset050'
            }}
            onClick={() => {
              handleSelect(title);
              setIsExpanded(!isExpanded);
              clickHandler(title);
            }}
            selected={isSelected === title}
          >
            {title}
          </MenuItem>
          <MenuDividerDropdown />
        </Fragment>
      ))}
    </>
  );
};
