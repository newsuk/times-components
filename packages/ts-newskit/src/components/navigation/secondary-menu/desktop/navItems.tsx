import React from 'react';
import { SecondaryMenuOptions, SecondaryMenuItem } from '../types';
import { StyledMenuItemsDesktop } from '../styles';

export const NavItems: React.FC<{
  options: SecondaryMenuOptions;
  data: SecondaryMenuItem[];
  hasMenuItem: number;
}> = ({ options, data, hasMenuItem }) => {
  const { handleSelect, isSelected } = options;

  return (
    <>
      {data.slice(0, hasMenuItem).map(({ title, url }) => (
        <StyledMenuItemsDesktop
          overrides={{
            paddingInlineStart: '6px',
            paddingInlineEnd: '6px',
            stylePreset: 'menuItemDesktop',
            typographyPreset: 'newPreset040'
          }}
          href={url}
          key={url}
          onClick={() => handleSelect(title)}
          selected={isSelected === title}
        >
          {title}
        </StyledMenuItemsDesktop>
      ))}
    </>
  );
};
