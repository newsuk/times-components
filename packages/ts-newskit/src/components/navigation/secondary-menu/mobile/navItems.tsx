import React from 'react';
import { NavItemsContainer, SecondaryNavMenuItemMob } from '../styles';
import { SecondaryMenuOptions, SecondaryMenuItem } from '../types';

export const NavItems: React.FC<{
  options: SecondaryMenuOptions;
  data: SecondaryMenuItem[];
}> = ({ options, data }) => {
  const { handleSelect, isSelected } = options;

  return (
    <NavItemsContainer>
      {data.map(({ url, slug, title }) => (
        <SecondaryNavMenuItemMob
          overrides={{
            paddingInline: 'space060',
            marginBlockStart: '-2px',
            marginInlineEnd: '-2px',
            stylePreset: 'secondaryMenuItem',
            typographyPreset: 'secondaryNavMenuItemMob'
          }}
          href={url}
          id={`vertical-${slug}`}
          onClick={() => handleSelect(slug)}
          key={slug}
          isSelected={isSelected === title}
        >
          {title}
        </SecondaryNavMenuItemMob>
      ))}
    </NavItemsContainer>
  );
};
