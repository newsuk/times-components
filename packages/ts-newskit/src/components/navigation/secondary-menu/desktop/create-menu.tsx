import React, { useEffect, useState, useRef } from 'react';
import { Menu } from 'newskit';
import { MenuContainer, Wrapper, MainMenu, StyledMenuSub } from '../styles';
import { SecondaryMenuOptions, SecondaryMenuItem } from '../types';
import { NavItems } from './navItems';
import { CreateMoreMenu } from './create-more-menu';
import { debounce, getWidth } from '../../../../utils';

export const CreateMenu: React.FC<{
  options: SecondaryMenuOptions;
  data: SecondaryMenuItem[];
}> = ({ options, data }) => {
  const contanierRef = useRef<HTMLDivElement>(null);
  const navListRef = useRef<HTMLDivElement>(null);
  const { isExpanded, setIsExpanded } = options;
  const [hasMenuItem, setHasMenuItem] = useState<number>(data.length);
  const [moreMenuItemsLength, setMoreMenuItemsLength] = useState<number>(0);
  const subMenuTitle = isExpanded ? 'Less' : 'More';

  useEffect(() => {
    const handleResize = async (navAdjustCount = 1) => {
      setMoreMenuItemsLength(moreMenuItemsLength);
      setHasMenuItem(hasMenuItem);

      const navListContainerWidth = await getWidth(contanierRef.current);
      const navListWidth = await getWidth(navListRef.current);

      if (
        navListWidth > navListContainerWidth - 200 &&
        navListWidth > 0 &&
        navListContainerWidth > 0
      ) {
        setMoreMenuItemsLength(moreMenuItemsLength + navAdjustCount);
        setHasMenuItem(hasMenuItem - navAdjustCount);

        const updatedNavListWidth = await getWidth(navListRef.current);
        updatedNavListWidth > navListContainerWidth - 200 &&
          handleResize(navAdjustCount + 1);
      }
    };

    window.addEventListener('resize', debounce(() => handleResize(), 500));
    handleResize();

    return () => {
      window.removeEventListener('resize', debounce(() => handleResize(), 500));
    };
  }, []);

  useEffect(
    () => {
      const checkIfClickedOutside = (e: any) => {
        if (
          isExpanded &&
          contanierRef.current &&
          !contanierRef.current.contains(e.target)
        ) {
          setIsExpanded(!isExpanded);
        }
      };
      document.addEventListener('click', checkIfClickedOutside);
      return () => {
        document.removeEventListener('click', checkIfClickedOutside);
      };
    },
    [isExpanded]
  );

  return (
    <MainMenu
      hasMoreItems={moreMenuItemsLength > 0 ? true : false}
      aria-label="Secondary Navigation"
      overrides={{
        spaceInline: 'space030'
      }}
      ref={contanierRef}
    >
      <Wrapper ref={navListRef}>
        <NavItems data={data} options={options} hasMenuItem={hasMenuItem} />
      </Wrapper>
      {moreMenuItemsLength > 0 && (
        <StyledMenuSub
          onClick={() => setIsExpanded(!isExpanded)}
          expanded={isExpanded}
          title={subMenuTitle}
          overrides={{
            paddingBlockStart: 'space020',
            paddingBlockEnd: 'space030',
            paddingInlineStart: 'space040',
            paddingInlineEnd: 'space040',
            stylePreset: `${isExpanded ? 'subMenuPreset2' : 'subMenuPreset1'}`,
            list: { stylePreset: 'subMenuItems' },
            typographyPreset: 'newPreset040'
          }}
          data-testid="more-sub-menu"
        >
          <MenuContainer>
            <Menu
              vertical
              overrides={{
                spaceInline: 'sizing000'
              }}
              aria-label="menu-multiple-auto"
            >
              <CreateMoreMenu
                data={data}
                options={options}
                moreMenuItemsLength={moreMenuItemsLength}
              />
            </Menu>
          </MenuContainer>
        </StyledMenuSub>
      )}
    </MainMenu>
  );
};
