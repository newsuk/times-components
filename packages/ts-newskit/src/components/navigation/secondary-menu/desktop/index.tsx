import React, { Fragment } from 'react';
import { MenuDivider } from '../styles';
import { SecondaryMenuOptions, SecondaryMenuItem } from '../types';
import { CreateMenu } from './create-menu';
import { useBreakpointKey } from 'newskit';

export const SecondaryNavDesktop: React.FC<{
  options: SecondaryMenuOptions;
  data: SecondaryMenuItem[];
  clickHandler: (title: string) => void;
}> = ({ options, data, clickHandler }) => {
  const breakpointKey = useBreakpointKey();

  return (
    <Fragment>
      <CreateMenu data={data} options={options} clickHandler={clickHandler} />
      <MenuDivider breakpointKey={breakpointKey} />
    </Fragment>
  );
};
