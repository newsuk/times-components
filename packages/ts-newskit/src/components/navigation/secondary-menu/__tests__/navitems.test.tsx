import React from 'react';
import { render } from '../../../../utils/test-utils';
import '@testing-library/jest-dom';
import { mainMenuItems } from '../fixtures/menu-items.json';
import { cleanup, fireEvent } from '@testing-library/react';
import { NavItems } from '../desktop/navItems';

const options = {
  handleSelect: jest.fn(),
  setIsExpanded: jest.fn(),
  isExpanded: false,
  isSelected: 'true'
};

const hasMenuItem = 3;
describe('Navitems Desktop', () => {
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it('should render snapshot', () => {
    const { asFragment } = render(
      <NavItems
        data={mainMenuItems}
        options={options}
        hasMenuItem={hasMenuItem}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render the menu item', () => {
    const { getByText } = render(
      <NavItems
        data={mainMenuItems}
        options={options}
        hasMenuItem={hasMenuItem}
      />
    );
    const title = getByText('Home');
    expect(title).toBeInTheDocument();
  });
  it('items should have ancher with href', () => {
    const { getAllByTestId } = render(
      <NavItems
        data={mainMenuItems}
        options={options}
        hasMenuItem={hasMenuItem}
      />
    );
    const title = getAllByTestId('buttonLink')[0];
    expect(title).toHaveAttribute('href', '/home');
  });

  it('items should have ancher with href', () => {
    const { getAllByTestId } = render(
      <NavItems
        data={mainMenuItems}
        options={options}
        hasMenuItem={hasMenuItem}
      />
    );
    const title = getAllByTestId('buttonLink')[0];
    fireEvent.click(title);
    expect(options.handleSelect).toHaveBeenCalled();
  });
});
