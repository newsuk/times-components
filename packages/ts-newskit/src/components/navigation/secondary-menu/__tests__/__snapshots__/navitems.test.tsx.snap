// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navitems Desktop should render snapshot 1`] = `
<DocumentFragment>
  <li
    class="nk-menu-item css-1kdinep"
  >
    <a
      aria-current="false"
      class="css-ilh6is"
      data-testid="buttonLink"
      href="/home"
    >
      <span
        class="css-t9dmcw"
      >
        Home
      </span>
    </a>
  </li>
  <li
    class="nk-menu-item css-1kdinep"
  >
    <a
      aria-current="false"
      class="css-ilh6is"
      data-testid="buttonLink"
      href="/news"
    >
      <span
        class="css-t9dmcw"
      >
        News
      </span>
    </a>
  </li>
  <li
    class="nk-menu-item css-1kdinep"
  >
    <a
      aria-current="false"
      class="css-ilh6is"
      data-testid="buttonLink"
      href="/comment"
    >
      <span
        class="css-t9dmcw"
      >
        Comment
      </span>
    </a>
  </li>
</DocumentFragment>
`;
